rootdev: Strip partitions less aggressively

vbatts opened the pull request:
In device names like dm-N or loopN, the N is not a partition number and
should not be stripped off.  These can arise for example if run inside a
chroot mounted on a loopback device.

BUG=chromium:748665, chromium:730144
TEST=New unit tests.

Change-Id: If7ecacbfcf00690376cb4ffc75baa45422579085
Reviewed-on: https://chromium-review.googlesource.com/585652
Commit-Ready: Benjamin Gordon <bmgordon@chromium.org>
Tested-by: Benjamin Gordon <bmgordon@chromium.org>
Reviewed-by: Simon Glass <sjg@chromium.org>
Reviewed-by: Mike Frysinger <vapier@chromium.org>
Signed-off-by: Vincent Batts <vbatts@kinvolk.io>

# pulling in upstream commit 

https://chromium.googlesource.com/chromiumos/third_party/rootdev/+/fc570685c1ca4acb871c706a66d0493b5e92c7c9

# How to use

[ describe what reviewers need to do in order to validate this PR ]


# Testing done

[Describe the testing you have done before submitting this PR. Please include both the commands you issued as well as the output you got.]

dongsupark reviewed (commented):
Thanks for digging into the old code of rootdev.
Code looks good. Tests run well.

However, the base branch should be `flatcar-master` instead of `master`.
Sorry, the default branch of the repo has been still `master` until now.
I have just set the default branch to `flatcar-master`, so it will be automatically set in further PRs.

Commit:
rootdev: Strip partitions less aggressively

In device names like dm-N or loopN, the N is not a partition number and
should not be stripped off.  These can arise for example if run inside a
chroot mounted on a loopback device.

BUG=chromium:748665, chromium:730144
TEST=New unit tests.

Change-Id: If7ecacbfcf00690376cb4ffc75baa45422579085
Reviewed-on: https://chromium-review.googlesource.com/585652
Commit-Ready: Benjamin Gordon <bmgordon@chromium.org>
Tested-by: Benjamin Gordon <bmgordon@chromium.org>
Reviewed-by: Simon Glass <sjg@chromium.org>
Reviewed-by: Mike Frysinger <vapier@chromium.org>
Signed-off-by: Vincent Batts <vbatts@kinvolk.io>

vbatts commented:
rebased on flatcar-master
